content/blog/2019/07/2019-07-30-jenkins-cli.adoc

```diff
@@ -9,14 +9,29 @@ author: linuxsuren
 
 As a developer, I usually use Jenkins like this:
 
-* Find a job which is related with your current work
+* Find a job which is related with my current work
 * Trigger that job
 * Check the output of log
 
-Sometimes, I might need to check the update center. Maybe a new plugin is needed, or update an exists plugin.
-Or, I want to upload a plugin from my computer. For all these cases, I just don't need a UI or even a browse.
+Sometimes, I might need to check the update center. Maybe a new plugin is needed, or I need to update an existing plugin.
+Or, I want to upload a plugin from my computer. For all these cases, I just don't need a UI or even a browser.
 I like to use a CLI to complete most of my tasks. For example, I use `kubectl` to manage my Kubernetes cluster,
-to create or modify the kubernetes resources. So, I start to think that why not using a CLI to manage my Jenkins.
+to create or modify the kubernetes resources. So, I start to think, 'Why not use a CLI to manage my Jenkins?'.
+
+== Why create a new one?
+
+First, I almost forgot about the existing Jenkins CLI, written in Java. Let me introduce how to use that one.
+
+Visit Jenkins page from `http://localhost:8080/jenkins/cli/`. You'll see a command like `java -jar jenkins-cli.jar -s http://localhost:8080/jenkins/ help`. So, a jar file needs to be download.
+We can use this command to complete this task `wget http://localhost:8080/jenkins/jnlpJars/jenkins-cli.jar`.
+
+Now you can see that this is not a Linux-style CLI. Please consider some points below:
+
+* The users must have a JRE. This is not convenient for developers who don't use Java.
+* Type more words. We always need to type `java -jar jenkins-cli.jar -s http://localhost:8080/jenkins/` as the initial command.
+* Cannot install it by some popular package manager, like `brew` or `yum`.
```

Comment:
Also the startup time is slow. Not sure if that was a factor for _you_.

Reply:
Thanks for your reminder. I'd like to move to next week.